controllers/rack.go

```diff
@@ -667,6 +691,20 @@ func (r *SingleClusterReconciler) upgradeRack(statefulSet *appsv1.StatefulSet, r
 			"rollingUpdateBatchSize", r.aeroCluster.Spec.RackConfig.RollingUpdateBatchSize,
 		)
 
+		// Create services for all pods if network policy is changed and rely on nodePort service.
```

Comment:
Now we have a common function upgradeOrRollingRestartRack, can you check if we can re-structure some code and carve out some common functionality from upgradeRack and rollingRestartRack function.

Reply:
`upgradeOrRollingRestartRack` func doesn't affect the functionality of upgradeRack and rollingRestartRack, so restructuring is not possible at the moment. However, I've moved the service creation code out in a func to be used for both